### PR TITLE
Dd/oracle

### PR DESCRIPTION
**Pull request by dafyddd**
1. use pyth_client instead of copy pasta code
2. allow pyth to be behind by 50 slots instead of the current 25 enforced by Pyth program

Tested on devnet (with devnet cfg flags removed) and works fine

### CODE REVIEW

**Comment by dafyddd** [comment]
@acamill Do you know why our CI always fails the UXD composability test?

File: program/src/processor.rs [hunk 1]
```diff
@@ -6667,15 +6663,19 @@ pub fn read_oracle(
...
             let conf = I80F48::from_num(price_account.agg.conf).checked_div(value).unwrap();
 
             #[cfg(not(feature = "devnet"))]
-            if price_account.agg.status != PriceStatus::Trading {
+            if price_account.agg.status != PriceStatus::Trading
+                && price_account.last_slot < curr_slot - PYTH_VALID_SLOTS
+            {
+                // Only ignore the pyth price if there hasn't been a valid slot in 50 slots
                 msg!("Pyth status invalid: {}", price_account.agg.status as u8);
                 return Err(throw_err!(MangoErrorCode::InvalidOraclePrice));
```
Comment (microwavedcola1) [inline program/src/processor.rs:6680]: nitpick (only fix if you feel like): I would make this Error mode specific so as to distinguish it from the other InvalidOraclePrice on line:6688

**Review by microwavedcola1 — APPROVED** [review]

**Comment by microwavedcola1** [comment]
> @acamill Do you know why our CI always fails the UXD composability test?

This fails due to github auth token to private repositories, @acamill  told us to ignore it for now, they will open source soon after audit and we wont require a token to run their tests

File: program/src/state.rs [hunk 2]
```diff
@@ -49,6 +49,7 @@ pub const FREE_ORDER_SLOT: u8 = u8::MAX;
 pub const MAX_NUM_IN_MARGIN_BASKET: u8 = 9;
 pub const INDEX_START: I80F48 = I80F48!(1_000_000);
 pub const PYTH_CONF_FILTER: I80F48 = I80F48!(0.10); // filter out pyth prices with conf > 10% of price
+pub const PYTH_VALID_SLOTS: u64 = 50; // Accept Pyth updates up to 50 slots behind
```
Comment (microwavedcola1) [inline program/src/state.rs:52]: nitpick: maybe add something from this discussion as a comment

**Comment by acamill** [comment]
> > @acamill Do you know why our CI always fails the UXD composability test?
> 
> This fails due to github auth token to private repositories, @acamill told us to ignore it for now, they will open source soon after audit and we wont require a token to run their tests

Oh yes I though you guys removed it entirely for now. 
I'll look into it today

**Comment by acamill** [comment]
Here is the PR #167

**Pull request merged**